crates/node/engine/src/task_queue/tasks/insert/task.rs

```diff
@@ -45,9 +47,65 @@ impl InsertTask {
     }
 
     /// Checks the response of the `engine_newPayload` call.
-    const fn check_new_payload_status(&self, status: &PayloadStatusEnum) -> bool {
+    const fn check_new_payload_status(status: &PayloadStatusEnum) -> bool {
         matches!(status, PayloadStatusEnum::Valid | PayloadStatusEnum::Syncing)
     }
+
+    fn try_into_block(
+        payload: OpExecutionPayload,
+        parent_beacon_block_root: FixedBytes<32>,
+    ) -> Result<OpBlock, InsertTaskError> {
+        match payload {
+            payload @ OpExecutionPayload::V1(_) => {
+                payload.try_into_block().map_err(InsertTaskError::FromBlockError)
+            }
+            payload @ OpExecutionPayload::V2(_) => {
+                payload.try_into_block().map_err(InsertTaskError::FromBlockError)
+            }
```

Comment:
Are we able to merge these two branches like so?

```suggestion
            payload @ OpExecutionPayload::V1(_) | payload @ OpExecutionPayload::V2(_) => {
                payload.try_into_block().map_err(InsertTaskError::FromBlockError)
            }
```